### PR TITLE
Treat relative paths in Cobertura reports as relative to the project …

### PR DESCRIPTION
**Pull request by abbeyj**
…base directory as opposed to the current directory.

This is intended to address #426.  That issue is currently marked closed but it is not fixed I've requested that it be reopened.

### CODE REVIEW

File: sonar-cxx-plugin/src/main/java/org/sonar/plugins/cxx/coverage/CoberturaParser.java [hunk 1]
```diff
@@ -37,6 +37,14 @@
  * {@inheritDoc}
  */
 public class CoberturaParser implements CoverageParser {
```
Comment (guwirth) [inline sonar-cxx-plugin/src/main/java/org/sonar/plugins/cxx/coverage/CoberturaParser.java]: @abbeyj looks like a specific solution for CoberturaParser. Could you move this ```baseDir``` to ```CoverageParser``` to make it more generic please.

File: sonar-cxx-plugin/src/main/java/org/sonar/plugins/cxx/coverage/CxxCoverageSensor.java [hunk 2]
```diff
@@ -63,7 +63,7 @@ public CxxCoverageSensor(Settings settings, FileSystem fs, ProjectReactor reacto
     super(settings, fs, reactor);
 
     this.reactor = reactor;
-    parsers.add(new CoberturaParser());
+    parsers.add(new CoberturaParser(fs.baseDir().getAbsolutePath()));
```
Comment (guwirth) [inline sonar-cxx-plugin/src/main/java/org/sonar/plugins/cxx/coverage/CxxCoverageSensor.java]: Would forward baseDir to all parsers in the ctor.

File: sonar-cxx-plugin/src/main/java/org/sonar/plugins/cxx/coverage/CoberturaParser.java (hunk 1 above)
Comment (abbeyj) [inline sonar-cxx-plugin/src/main/java/org/sonar/plugins/cxx/coverage/CoberturaParser.java]: CoverageParser is an interface so I can't move it there.  Do you want me to create a base class and put baseDir there?  If so, do you have a preference for a name?

**Comment by guwirth** [comment]
@abbeyj Sorry for the late answer. Would do it like here https://github.com/wenns/sonar-cxx/blob/master/sonar-cxx-plugin/src/main/java/org/sonar/plugins/cxx/utils/CxxReportSensor.java

New class CxxCoverageParser which implements CoverageParser. Current location of your class in coverage folder is ok.

**Comment by guwirth** [comment]
Thx

**Pull request merged**